cpp/src/DataStorm/NodeI.cpp

```diff
@@ -199,32 +197,12 @@ NodeI::createSession(
             {
                 if (session->checkSession())
                 {
-                    if (isWellKnown && current.con && session->getConnection() != current.con)
-                    {
-                        // If the peer is using a well-known proxy and requests a session using a different connection,
-                        // assume the current session connection is being closed and reconnect using the new connection.
-                        //
-                        // Otherwise, once the current connection is actually closed, we won't be able to reconnect.
-                        if (traceLevels->session > 2)
-                        {
-                            Trace out(traceLevels->logger, traceLevels->sessionCat);
-                            out << "node '" << current.id << "' is discarding current session '" << session->getId()
-                                << "' because subscriber '" << subscriber
-                                << "' (using a well-known proxy) requested creating a session over a new connection";
-                        }
-                        session->disconnect();
-                        session->reconnect(*subscriber, current.con);
-                    }
```

Comment:
Here `disconnected()` marked the local side as disconnected, and then the call to reconnect starts the recovery. The assumption was that at this point the peer was disconnected and that is why `createSession` request is received over a new connection. The assumption is bogus for example a concurrent session attempt might have just success over a different connection.